common-content/en/module/tools/first-nodejs-program/index.md

```diff
@@ -0,0 +1,114 @@
++++
+title = "Writing a NodeJS program"
+headless = true
+time = 60
+facilitation = false
+emoji= "🛠️"
+hide_from_overview = true
+objectives = [
+    "Write a zero-dependencies NodeJS program",
+]
++++
+
+Below we have a small NodeJS program. It is a bit like `wc`. It counts words in a file. Specifically, it counts words which contain a hyphen (`-`) character.
+
+It accepts one command line argument - the path of the file to read and count.
+
+Its output to stdout is just the number of words which contain a hyphen.
+
+It uses the same language (JavaScript) as we've written before, but uses some different APIs.
+
+```js
+import process from "node:process";
+import { promises as fs } from "node:fs";
+
+const argv = process.argv.slice(2);
+if (argv.length != 1) {
+    console.error(`Expected exactly 1 argument (a path) to be passed but got ${argv.length}.`);
+    process.exit(1);
+}
+const path = argv[0];
+
+const content = await fs.readFile(path, "utf-8");
+const wordsContainingHyphens = content.split(" ").filter((word) => word.indexOf("-") > -1).length;
```

Comment:
```suggestion
const wordsContainingHyphens = content.split(" ").filter((word) => word.includes("-")).length;
```

Reply:
Really nitpicky, but `wordsContaininingHyphens` implies that it's the actual list of words, but this is actually the count of words.

Reply:
Updated to default to counting words containing "e"s :) And renamed the var slightly clumsily